### PR TITLE
chore: Implement nix flake pattern to build package

### PR DESCRIPTION
**Pull request by jonybur**
# Description

Builds the wasm flake on noir. Updates the workflow so that it only runs when dispatched.

Resolves https://github.com/orgs/noir-lang/projects/1/views/13?pane=issue&itemId=29978086

# PR Checklist\*

- [X] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: .github/workflows/update.yml [hunk 1]
```diff
@@ -6,8 +6,9 @@ on:
       noir-ref:
         description: The noir reference to checkout
         required: false
-  schedule:
-    - cron: "0 2 * * *" # run at 2 AM UTC
+  push:
+    branches:
```
Comment (kobyhallx) [inline .github/workflows/update.yml]: Should this be removed before we pull into master?

File: .github/workflows/update.yml [hunk 2]
```diff
@@ -48,9 +49,19 @@ jobs:
           name: nargo-cache
           authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
 
-      - name: Build environment and Compile
+      - name: Build with Nix
         run: |
-          nix-shell noir_wasm.nix --pure --run "./build.sh"
+          nix build -L github:noir-lang/noir/jb/noir-wasm-workflow#wasm --extra-experimental-features nix-command --extra-experimental-features flakes
```
Comment (kobyhallx) [inline .github/workflows/update.yml]: Are `--extra-experimental-features nix-command --extra-experimental-features flakes` needed? They should be removed...

File: .github/workflows/update.yml [hunk 3]
```diff
@@ -59,26 +70,29 @@ jobs:
...
-        run: |
-          npm publish --tag latest
-        env:
-          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+      # - name: Publish to npm (latest tag)
+      #   if: ${{ inputs.noir-ref }}
+      #   run: |
+      #     npm publish --tag latest
+      #   env:
+      #     NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (kobyhallx) [inline .github/workflows/update.yml]: We do want these to be uncommented on master?

File: .github/workflows/update.yml [hunk 4]
```diff
@@ -59,26 +70,29 @@ jobs:
...
-      - name: Setup NodeJS
-        uses: actions/setup-node@v3
-        with:
-          node-version: "18"
-          registry-url: "https://registry.npmjs.org"
+      # - name: Setup NodeJS
+      #   uses: actions/setup-node@v3
+      #   with:
+      #     node-version: "18"
+      #     registry-url: "https://registry.npmjs.org"
```
Comment (kobyhallx) [inline .github/workflows/update.yml]: NodeJS setup should not be needed as runner already has it?

File: .github/workflows/update.yml (hunk 2 above)
Comment (jonybur) [reply .github/workflows/update.yml]: Removed

File: .github/workflows/update.yml (hunk 1 above)
Comment (jonybur) [reply .github/workflows/update.yml]: Removed

**Review by kobyhallx — APPROVED** [review]
👍

**Pull request merged**